pop-api/src/v0/nfts.rs

```diff
@@ -211,3 +213,161 @@ pub(crate) enum NftCalls {
+    BadWitness,
+    /// Collection ID is already taken.
+    CollectionIdInUse,
+    /// Items within that collection are non-transferable.
+    ItemsNonTransferable,
+    /// The provided account is not a delegate.
+    NotDelegate,
+    /// The delegate turned out to be different to what was expected.
+    WrongDelegate,
+    /// No approval exists that would allow the transfer.
+    Unapproved,
+    /// The named owner has not signed ownership acceptance of the collection.
+    Unaccepted,
+    /// The item is locked (non-transferable).
+    ItemLocked,
+    /// Item's attributes are locked.
+    LockedItemAttributes,
+    /// Collection's attributes are locked.
+    LockedCollectionAttributes,
+    /// Item's metadata is locked.
+    LockedItemMetadata,
+    /// Collection's metadata is locked.
+    LockedCollectionMetadata,
+    /// All items have been minted.
+    MaxSupplyReached,
+    /// The max supply is locked and can't be changed.
+    MaxSupplyLocked,
+    /// The provided max supply is less than the number of items a collection already has.
+    MaxSupplyTooSmall,
+    /// The given item ID is unknown.
+    UnknownItem,
+    /// Swap doesn't exist.
+    UnknownSwap,
+    /// The given item has no metadata set.
+    MetadataNotFound,
+    /// The provided attribute can't be found.
+    AttributeNotFound,
+    /// Item is not for sale.
+    NotForSale,
+    /// The provided bid is too low.
+    BidTooLow,
+    /// The item has reached its approval limit.
+    ReachedApprovalLimit,
+    /// The deadline has already expired.
+    DeadlineExpired,
+    /// The duration provided should be less than or equal to `MaxDeadlineDuration`.
+    WrongDuration,
+    /// The method is disabled by system settings.
+    MethodDisabled,
+    /// The provided setting can't be set.
+    WrongSetting,
+    /// Item's config already exists and should be equal to the provided one.
+    InconsistentItemConfig,
+    /// Config for a collection or an item can't be found.
+    NoConfig,
+    /// Some roles were not cleared.
+    RolesNotCleared,
+    /// Mint has not started yet.
+    MintNotStarted,
+    /// Mint has already ended.
+    MintEnded,
+    /// The provided Item was already used for claiming.
+    AlreadyClaimed,
+    /// The provided data is incorrect.
+    IncorrectData,
+    /// The extrinsic was sent by the wrong origin.
+    WrongOrigin,
+    /// The provided signature is incorrect.
+    WrongSignature,
+    /// The provided metadata might be too long.
+    IncorrectMetadata,
+    /// Can't set more attributes per one call.
+    MaxAttributesLimitReached,
+    /// The provided namespace isn't supported in this call.
+    WrongNamespace,
+    /// Can't delete non-empty collections.
+    CollectionNotEmpty,
+    /// The witness data should be provided.
+    WitnessRequired,
+}
+
+impl From<u32> for Error {
+    fn from(value: u32) -> Self {
+        use Error::*;
+        match value {
```

Comment:
Using `num_derive` & `num_traits` would remove the need to do these manual matches.

Reply:
I'm personally hesitant to take a dependency when it can be implemented perfectly easily in local Rust code. Suggest we revist when the number of errors explode.